src/main/java/com/uber/cadence/internal/worker/LocalActivityPollTask.java

```diff
@@ -0,0 +1,39 @@
+/*
+ *  Copyright 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Modifications copyright (C) 2017 Uber Technologies, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not
+ *  use this file except in compliance with the License. A copy of the License is
+ *  located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed on
+ *  an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package com.uber.cadence.internal.worker;
+
+import java.util.Queue;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.function.Consumer;
+import org.apache.thrift.TException;
+
+final class LocalActivityPollTask
+    implements Poller.PollTask<LocalActivityWorker.Task>, Consumer<LocalActivityWorker.Task> {
+  private static final int QUEUE_SIZE = 1000;
+  private Queue<LocalActivityWorker.Task> pendingTasks = new ArrayBlockingQueue<>(QUEUE_SIZE);
+
+  @Override
+  public LocalActivityWorker.Task poll() throws TException {
+    return pendingTasks.poll();
+  }
+
+  @Override
+  public void accept(LocalActivityWorker.Task task) {
+    pendingTasks.offer(task);
```

Comment:
What is the behavior when the queue is full?

Reply:
I added timeout in the latest version. When it times out we will do similar things as heartbeating (i.e. sending all the results we have back to server and ask for a decision task immediately).